WcaOnRails/app/models/competition.rb

```diff
@@ -427,16 +439,24 @@ def dangerously_close_to?(c)
     self.kilometers_to(c) <= NEARBY_DISTANCE_KM_DANGER && days_until.abs < NEARBY_DAYS_DANGER
   end
 
+  def results_uploaded?
+    Result.exists?(competitionId: self.id)
+  end
+
   def user_can_view?(user)
     self.showAtAll || (user && user.can_manage_competition?(self))
   end
 
+  def in_progress?
+    start_date <= Date.today && end_date >= Date.today
```

Comment:
Nit: I like to write these "squeeze conditions" so it's more obvious what's getting bounded. See:

```ruby
start_date <= Date.today && Date.today <= end_date
```

This is closer to python's *awesome* syntax: `start_date <= Date.today <= end_date`, which I really wish ruby supported.